lucene/core/src/java/org/apache/lucene/util/IntroSelector.java

```diff
@@ -33,26 +32,87 @@ public final void select(int from, int to, int k) {
     quickSelect(from, to, k, maxDepth);
   }
 
-  // heap sort
-  // TODO: use median of median instead to have linear worst-case rather than
-  // n*log(n)
-  void slowSelect(int from, int to, int k) {
-    new Sorter() {
+  int slowSelect(int from, int to, int k) {
+    return medianOfMediansSelect(from, to-1, k);
+  }
+
+  int medianOfMediansSelect(int from, int to, int k) {
```

Comment:
It took me a while to figure out that `to` is inclusive here while it is exclusive elsewhere in this file (e.g. in `select` and `quickselecet`). Maybe rename it to `toInclusive` or something along those lines to make it clear? Or maybe call them `left` and `right` like you did in some other methods?